oci/archive/oci_src.go

```diff
@@ -0,0 +1,153 @@
+package archive
+
+import (
+	"context"
+	"io"
+
+	"os"
+
+	ocilayout "github.com/containers/image/oci/layout"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/pkg/archive"
+	digest "github.com/opencontainers/go-digest"
+	imgspecv1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type ociArchiveImageSource struct {
+	ref         ociArchiveReference
+	unpackedSrc types.ImageSource
+	tempDir     string
+}
+
+var tempDirRef *tempDirOciRef
```

Comment:
Okay, I got rid of the global variable and added this to the `ociArchiveReference` struct. However, I ran into an issue that I have a temporary fix for.

`LoadManifestDescriptor` needs to be able to access the image name from the index.json for the `kpod load` command. Since `LoadManifestDescriptor` is called before `newImageSource` for `kpod load`, I tried to check if `tempDirRef` has a value in the instance of `ociArchiveReference` and setting it in one of the functions. However, two instances were being created and `ociArchiveReference` wouldn't retain its value.

I can't seem to figure out why the instance of `ociArchiveReference` is not the same - could be that `LoadManifestDescriptor` is being called from `libkpod/copy.go`. I have a temporary fix in place that just creates a temp directory and deletes it in `LoadManifestDescriptor`. 

@mtrmac I know this might not be the ideal way, but just wanted your feedback and advice on this issue.